src/parser.jl

```diff
@@ -32,9 +35,29 @@ function function_body_lines!(flines, ast::Expr, infunction)
     else
         args = ast.args
     end
-    infunction |= isfuncexpr(ast)
-    for arg in args
-        flines = function_body_lines!(flines, arg, infunction)
+
+    # Check whether we are looking at a function declaration.
+    # Then also make sure we are not looking at a function declaration
+    # that does not also define a method by checking the length of
+    # ast.args and making sure it is at least length==2. The test for
+    # Expr might not be necessary but also can't harm. Note that this works
+    # for both function declarations that use the `function` keyword, and
+    # declarations of the form `foo() = expr`, in both cases the method
+    # body ends up being `ast.args[2]` (if it exists).
+    if isfuncexpr(ast) && length(ast.args)>=2 && ast.args[2] isa Expr
+        # Only look in function body and ignore the function signature
+        # itself. Sometimes function signatures have line nodes inside
+        # and we don't want those lines to be identified as runnable code.
+        # In this context, ast.args[1] is the function signature and
+        # ast.args[2] is the method body
+        for arg in ast.args[2].args
+            flines = function_body_lines!(flines, ast.arg, true)
```

Comment:
```suggestion
            flines = function_body_lines!(flines, arg, true)
```

Reply:
Ah, thanks, that was a stupid regression :)